CBOR_DECODER_NEDATA: return required bytes

mcnulty opened the pull request:
This PR adds the minimum number of bytes required by cbor_stream_decode to continue parsing when that function returns CBOR_DECODER_NEDATA as the status in the cbor_decoder_result.

This is useful when using a byte stream transport such as a raw TCP socket or UNIX pipe and blocking I/O.

I'm not sure if this is the best way to expose this information in the API, but it seemed to do the trick.

PJK commented:
Hi @mcnulty, thank you for the PR, I agree that this is a useful API extension. The implementation looks good, but it should also come with some tests and documentation (probably here http://libcbor.readthedocs.io/en/latest/api/decoding.html#_CPPv29cbor_load9cbor_data6size_tP16cbor_load_result). 

If you want to add those, that would be great, if not, I will get back to this in a few days

mcnulty commented:
I added a basic test case for the behavior of the `required` field in the `cbor_decoder_result` struct to `cbor_stream_decode_test`. I also enhanced `assert_decoder_result` to ensure that the `required` field is 0 for all existing usages by other tests.

As far as documentation, I added a description of the `required` field to the `cbor_decoder_result` struct. I think the description will show up in the docs for [cbor_decoder_result](http://libcbor.readthedocs.io/en/v0.5.0/streaming/decoding.html#_CPPv219cbor_decoder_result), right? Would this be a sufficient addition to the docs for this feature or were you looking for something more?

PJK commented:
This is great, in it goes :) Thank you!
